integration_tests/index.test.ts

```diff
@@ -0,0 +1,149 @@
+import { Kepler, authenticator } from 'kepler-sdk';
+import { DAppClient } from '@airgap/beacon-sdk';
+import { InMemorySigner } from '@taquito/signer';
+import { b58cencode, prefix } from "@taquito/utils";
+import { DockerComposeEnvironment, StartedDockerComposeEnvironment } from "testcontainers";
+import { ContractClient } from "orbit-manifest";
+
+import path = require('path');
+const fetch = require('node-fetch');
+import crypto = require('crypto');
+
+const secrets = ['edsk3QoqBuvdamxouPhin7swCvkQNgq4jP5KZPbwWNnwdZpSpJiEbq', 'edsk3RFfvaFaxbHx8BMtEW1rKQcPtDML3LXjNqMNLCzC3wLC1bWbAt'];
+
+const buildContext = path.resolve(__dirname, '..');
+const composeFile = 'sandbox.yml';
+const kepler1Port = 8000;
+const kepler2Port = 9000;
+
+const genClient = async (secret: string = b58cencode(
+    crypto.randomBytes(32),
+    prefix.edsk2
+)): Promise<DAppClient> => {
+    const ims = new InMemorySigner(secret);
+    // @ts-ignore
+    return {
+        // @ts-ignore
+        getActiveAccount: async () => ({ publicKey: await ims.publicKey(), address: await ims.publicKeyHash() }),
+        // @ts-ignore
+        requestSignPayload: async ({ payload }) => ({ signature: await ims.sign(payload).then(res => res.prefixSig) })
+    }
+}
+
+type HostInfo = {
+    id: string,
+    port: number,
+    url: string
+}
+
+const create = async (
+    secret: string,
+    [host, ...peers]: HostInfo[],
+) => {
+    // get IDs and multiaddrs of all kepler nodes
+    const hostInfo = await Promise.all([host, ...peers].map(async host => {
+        const api = "http://" + host.url + ":" + host.port;
+        // @ts-ignore
+        const key: string = await fetch(api + "/host").then(r => r.json()).then(({ id }) => id)
+        return { [key]: ["/ip4/0.0.0.0/tcp/" + (host.port + 1)] }
+    })).then(hosts => hosts.reduce((h, acc) => ({ ...h, ...acc })));
+
+    // deploy contract
+    const client = new ContractClient({
+        tzktBase: "http://localhost:5000",
+        nodeURL: "http://localhost:8732",
+        contractType: "",
+        signer: {
+            type: 'secret',
+            secret
+        }
+    });
+
+    const contract = await client.originate({
+        hosts: hostInfo,
+        admins: []
+    });
+
+    // wait for tzkt to index the contract
+    await new Promise(resolve => setTimeout(resolve, 1000));
```

Comment:
This can be trusted? Is the test reliable with this background operation?

Reply:
The test is reliable with much smaller timeouts, but at least a small wait is required for the contract origination to be completed by the `flextesa` node. In practice/real world, this will probably be at least 1 tezos block (30 seconds) and probably more if you want to be safe ([since grenada, this means about 2-3 blocks](https://blog.nomadic-labs.com/faster-finality-with-emmy.html)).